tests/rptest/services/keycloak.py

```diff
@@ -196,25 +259,58 @@ def login_admin_user(self, node, username, password):
 
     def generate_oauth_config(self, node, client_id):
         secret = self.admin.get_client_secret(client_id)
-        token_endpoint = self.get_token_endpoint(node)
-        return OAuthConfig(client_id, secret, token_endpoint)
+        use_ssl = self.tls_provider is not None
+        ca_cert = self.tls_provider.ca.crt if self.tls_provider is not None else None
+        token_endpoint = self.get_token_endpoint(node, use_ssl=use_ssl)
+        return OAuthConfig(client_id, secret, token_endpoint, ca_cert=ca_cert)
 
     def pids(self, node):
         return node.account.java_pids('quarkus')
 
     def alive(self, node):
         return len(self.pids(node)) > 0
 
+    def setup_tls(self, node) -> tuple[str | None, str | None, str | None]:
+        if self.tls_provider is None:
+            return None, None, None
+        if self._is_cdt:
+            return SYS_TLS_CRT_FILE, SYS_TLS_KEY_FILE, SYS_TLS_TRUST_STORE_FILE
+        else:
+            cert = self.tls_provider.create_broker_cert(self, node)
+            node.account.copy_to(cert.crt, KC_TLS_CRT_FILE)
+            node.account.copy_to(cert.key, KC_TLS_KEY_FILE)
+            node.account.copy_to(cert.ca.crt, KC_TLS_TRUST_STORE_FILE)
```

Comment:
Should these and any other files be cleanup up in `clean_node(self)`, which I think is a member of  `ducktape.services.service.Service`.

Reply:
Probably!

Reply:
done